### ISSUE TITLE
[Doc] Add missing indicators of supported queue type on Logstash settings

### ISSUE THREAD

**Issue by a03nikki** [elastic/logstash#12536]
Similar to https://github.com/elastic/logstash/issues/10999, it would be helpful if the documentation called out which queue types the queue settings apply to in all cases.

For example, the [pipeline.yml](https://github.com/elastic/logstash/blob/master/config/pipelines.yml#L17) and [logstash.yml](https://github.com/elastic/logstash/blob/master/config/logstash.yml#L158) files both include in the comments the `queue.type` they apply to.  However, the related documentation pages do not always include this information.

For example on doc page [logstash.yml](https://www.elastic.co/guide/en/logstash/7.10/logstash-settings-file.html), the `queue.max_bytes` setting does not indicate that it is only for persistent queues

> queue.max_bytes | The total capacity of the queue in  number of bytes. Make sure the capacity of your disk drive is greater  than the value you specify here. If both queue.max_events and queue.max_bytes are specified, Logstash uses whichever criteria is reached first. | 1024mb (1g)
> -- | -- | --

However, there are other settings do include these details, such as

> queue.checkpoint.acks | The maximum number of ACKed events before forcing a checkpoint when persistent queues are enabled (`queue.type: persisted`). Specify queue.checkpoint.acks: 0 to set this value to unlimited.
> -- | --

In the example of `queue.max_bytes`, the only reason why I went digging for more information was from this sentence on [Persistent Queues](https://www.elastic.co/guide/en/logstash/7.10/persistent-queues.html) page

> By default, Logstash uses in-memory bounded queues between pipeline stages (inputs → pipeline workers) to buffer events. The size of these in-memory queues is fixed and not configurable.

**Comment by kaisecheng** [comment]
@karenzone I would like to have your thought on this issue. The author has confusion about the queue options on this [page](https://www.elastic.co/guide/en/logstash/current/persistent-queues.html#configuring-persistent-queues). She doesn't notice all options are only available for persistent queue, not memory queue. She points out the [source code](https://github.com/elastic/logstash/blob/main/config/logstash.yml#L200-L244) that has written down the queue type explicitly in each option.

The current doc is very clear to me that the options are for PQ, so I do not understand her confusion. Do you think we miss anything?

**Comment by karenzone** [comment]
This issue is from 2020, and we've done quite a bit of rework on queues since then.  I'll reach out to see if the rework alleviates confusion, or if we have more work to do. Thanks for the ping.

**Comment by karenzone** [comment]
Hi @a03nikki 👋  We've done some rework on the [queue information](https://www.elastic.co/guide/en/logstash/current/resiliency.html) in the documentation recently, splitting out sections for [mem queue](https://www.elastic.co/guide/en/logstash/current/memory-queue.html) and [persistent queue](https://www.elastic.co/guide/en/logstash/current/persistent-queues.html).  Please take a look, and let me know if the recent changes have addressed your concerns or if there is more we can do.

**Comment by a03nikki** [comment]
@karenzone : You are correct that I opened this against an older version of the documentation (7.10 based on the links in description). 

I have reviewed the latest version (8.2) and the pages are more clear. I appreciate the dedicated ["Memory queue"](https://www.elastic.co/guide/en/logstash/8.2/memory-queue.html) page that was added and settings added to the ["Persistent queues (PQ)"](https://www.elastic.co/guide/en/logstash/8.2/persistent-queues.html) page.

There are a few settings on the logstash.yml doc page that were missing the "(queue.type: persisted)" so I opened #14127 for them.
